### ISSUE TITLE
Move `max distance` to the joint rule components

### ISSUE THREAD

**Issue by chenkasirer** [gramaziokohler/compas_timber#182]
Rational is that e.g. L topology shouldn't allow centerlines which don't meet to be considered an L

**Comment by chenkasirer** [comment]
Been playing with this a little. I think we should have some kind (ideally quite small) of a global limit on how high this, maybe something that's proportional to the average beam dims or maybe to the unit used. Once this gets big enough funky things start happening and then the joint themselves have to deal with some really sick cases, potentially making the code less readable. 

@obucklin @jonashaldemann @funkchaser what do you think?

**Comment by chenkasirer** [comment]
rule component max distance overrides the one of the assembly component's for that specific rule.
assembly max distance gets some default when nothing specified

**Comment by chenkasirer** [comment]
cap `max_distance` to something proportional to the unit used
https://github.com/orgs/gramaziokohler/projects/5/views/1?pane=issue&itemId=51489076

**Comment by jonashaldemann** [comment]
Maybe it could be the largest width or height of all beams in the Assembly. You never want it any bigger than that.

**Comment by jonashaldemann** [comment]
![image](https://github.com/gramaziokohler/compas_timber/assets/104436733/82537417-eaac-42a6-99ae-85b127d1b5eb)

**Comment by chenkasirer** [comment]
IDEA: 
- `max_distance` is set globally in the Assembly component and is used to find potential pairs
- each Joint also has a `max_distance` setting which may be lower, this allows the joint itself to check again and potentially reject this pair

**Comment by jonashaldemann** [comment]
yes, I think that could help to customize and solve some joints.
